pkg/client/cache/reflector.go

```diff
@@ -233,11 +233,6 @@ loop:
 		}
 	}
 
-	watchDuration := time.Now().Sub(start)
-	if watchDuration < 1*time.Second && eventCount == 0 {
-		glog.V(4).Infof("Unexpected watch close - watch lasted less than a second and no items received")
-		return errors.New("very short watch")
-	}
```

Comment:
Why are you removing this block? The intention--broken now, I see--is that this error ought to be treated like the connection never opened in the first place, for the purpose of preventing a hot retry loop.